engine/src/configuration/state.cc

```diff
@@ -38,13 +37,37 @@ struct setter : public setter_base {
   bool apply_from_cfg(state& obj, char const* value) override {
     try {
       U val(0);
-      if (!string::to(value, val))
-        return false;
-      (obj.*ptr)(val);
-    } catch (std::exception const& e) {
-      SPDLOG_LOGGER_ERROR(config_logger, "fail to update {} with value {}: {}",
-                          setter_base::_field_name, value, e.what());
-      return false;
+      if constexpr (std::is_same_v<U, bool>) {
+        if (!absl::SimpleAtob(value, &val))
+          return false;
+        (obj.*ptr)(val);
+      } else if constexpr (std::is_same_v<U, uint16_t>) {
+        uint32_t v;
+        if (!absl::SimpleAtoi(value, &v))
+          return false;
+        if (v > 0xffffu)
+          return false;
+        else
+          val = v;
+        (obj.*ptr)(val);
+      } else if constexpr (std::is_integral<U>::value) {
+        if (!absl::SimpleAtoi(value, &val))
+          return false;
+        (obj.*ptr)(val);
+      } else if constexpr (std::is_same_v<U, float>) {
+        if (!absl::SimpleAtof(value, &val))
+          return false;
+        (obj.*ptr)(val);
+      } else if constexpr (std::is_same_v<U, double>) {
+        if (!absl::SimpleAtod(value, &val))
+          return false;
+        (obj.*ptr)(val);
+      } else {
+        static_assert(std::is_integral_v<U> || std::is_floating_point_v<U> ||
+                      std::is_same_v<U, bool>);
+      }
+    } catch (const std::exception& e) {
+      config_logger->error(e.what());
```

Comment:
a better log like config_logger->error("fail to parse {}={}: {}", setter_base::_field_name, value, e.what());